### PR TITLE
ci: fix github actions yaml spec errors

### PR DESCRIPTION
**Pull request by Serpentiel**
fixes GitHub Actions YAML spec errors, also sets Go version to use `go.mod` everywhere

### CODE REVIEW

File: .github/workflows/acceptance-tests.yaml [hunk 1]
```diff
@@ -10,7 +10,7 @@ jobs:
       - uses: actions/checkout@v3
       - uses: actions/setup-go@v4
         with:
-          go-version: '1.18'
+          go-version: stable
```
Comment (byashimov) [inline .github/workflows/acceptance-tests.yaml:13]: I haven't tested 1.20. Please bring back 1.19

File: .github/workflows/pages.yaml [hunk 2]
```diff
@@ -1,10 +1,10 @@
 name: pages
 
 on:
-  workflow_dispatch:
+  workflow_dispatch: {}
```
Comment (byashimov) [inline .github/workflows/pages.yaml:4]: Empty object, why?

File: .github/workflows/release.yaml [hunk 3]
```diff
@@ -54,6 +54,6 @@ jobs:
       with:
         push: true
         platforms: linux/amd64,linux/arm64
-        tags: aivenoy/aiven-operator:${{ steps.action_vars.outputs.TAG }}
+        tags: aivenoy/aiven-operator:${{ steps.action_vars.outputs.tag }}
         build-args: |
-          VERSION=${{steps.action_vars.outputs.TAG}}
+          VERSION=${{ steps.action_vars.outputs.tag }}
```
Comment (byashimov) [inline .github/workflows/release.yaml:59]: Please have tested this.

File: .github/workflows/tests.yaml [hunk 4]
```diff
@@ -1,7 +1,7 @@
 name: tests
 
 on:
-  workflow_dispatch:
+  - workflow_dispatch
```
Comment (byashimov) [inline .github/workflows/tests.yaml:4]: Let's have `on` of the same style across all workflows. It is a list here and an object in `pages`. Worth to mention, it is an object in [docs](https://docs.github.com/en/actions/using-workflows/triggering-a-workflow).

File: .github/workflows/tests.yaml [hunk 5]
```diff
@@ -13,10 +13,10 @@ jobs:
       - uses: actions/checkout@v3
       - uses: actions/setup-go@v4
         with:
-          go-version: '1.19'
+          go-version: stable
```
Comment (byashimov) [inline .github/workflows/tests.yaml:16]: Please revert.

**Review by byashimov — CHANGES_REQUESTED** [review]

File: .github/workflows/pages.yaml (hunk 2 above)
Comment (Serpentiel) [reply .github/workflows/pages.yaml:4]: YAML spec

File: .github/workflows/release.yaml (hunk 3 above)
Comment (Serpentiel) [reply .github/workflows/release.yaml:59]: it changes the name

File: .github/workflows/tests.yaml (hunk 4 above)
Comment (Serpentiel) [reply .github/workflows/tests.yaml:4]: GitHub's official linter suggests this to be an array when the object functionality is not used

File: .github/workflows/tests.yaml (hunk 5 above)
Comment (Serpentiel) [reply .github/workflows/tests.yaml:16]: see first response

File: .github/workflows/tests.yaml (hunk 4 above)
Comment (byashimov) [reply .github/workflows/tests.yaml:4]: Then please change it for pages.

> Let's have on of the same style across all workflows. It is a list here and an object in pages.

File: .github/workflows/acceptance-tests.yaml (hunk 1 above)
Comment (byashimov) [reply .github/workflows/acceptance-tests.yaml]: Then what it does? [Documentation](https://github.com/actions/setup-go#getting-go-version-from-the-gomod-file) says you should use different option then. And for `go-version`:

> go-version: '1.18.0-rc.1' # The Go version to download (if necessary) and use.
> If stable is provided, action will get the latest stable version from the [go-versions](https://github.com/actions/go-versions/blob/main/versions-manifest.json) repository manifest.

Comment (Serpentiel) [reply .github/workflows/acceptance-tests.yaml]: I have provided an answer to about this during our internal discussion, but here's a follow-up either way: changing the `go-version` from `1.18` to `stable` in this workflow doesn't actually change the version of Go being used—instead, it sets the Go version to be the latest `stable` release, while still respecting the [Go 1 compatibility promise](https://go.dev/doc/go1compat)—this means that any valid Go program written under Go 1.x should continue to work without modification under any future Go 1.y release; additionally, packages and programs compiled with earlier versions of Go will continue to work with future versions of the Go compiler as the Go project has made a commitment to maintain the stability of the language and its standard library, so developers can rely on their code continuing to work as expected even as new versions of Go are released

Comment (Serpentiel) [reply .github/workflows/acceptance-tests.yaml]: changed to `go-version-file: go.mod`

File: .github/workflows/tests.yaml (hunk 4 above)
Comment (Serpentiel) [reply .github/workflows/tests.yaml]: I've standardized this across all the workflows 🎉

**Review by byashimov — APPROVED** [review]
lgtm

**Pull request merged**